CWE Top 25 2019 & 2020

shaopeng-gh opened the pull request.

shaopeng-gh committed:
1st Draft

shaopeng-gh committed:
CWE Top 25 2019 and 2020

shaopeng-gh committed:
Merge branch 'users/shaopeng-gh/cwetop25' of https://github.com/sarif-standard/taxonomies into users/shaopeng-gh/cwetop25

shaopeng-gh committed:
Merge branch 'users/shaopeng-gh/cwetop25' of https://github.com/sarif-standard/taxonomies into users/shaopeng-gh/cwetop25

shaopeng-gh committed:
Merge branch 'users/shaopeng-gh/cwetop25' of https://github.com/sarif-standard/taxonomies into users/shaopeng-gh/cwetop25

eddynaka commented on Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs:
```diff
@@ -139,10 +139,10 @@ private Run ConvertToSarif(List<CweCsvRecord> records, string version)
-                ReleaseDateUtc = Constants.CWE.ReleaseDate,
+                ReleaseDateUtc = Constants.CWE_Comprehensive_V43.ReleaseDate,
                 InformationUri = new Uri("https://cwe.mitre.org/data/published/cwe_v4.4.pdf"),
                 DownloadUri = new Uri("https://cwe.mitre.org/data/xml/cwec_v4.4.xml.zip"),
```
>DownloadUri [](http://example.com/codeflow?start=16&length=11)

this does not look right.
The version is 43, but its point to 44 #Closed

shaopeng-gh replied:
forgot to change the url, fixed

shaopeng-gh committed:
fix the wrong url

eddynaka commented:
>         private Run ConvertToSarif(List<CweCsvRecord> records, string version)

should we delete this? i dont see any usage for the csv method

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293)

---
Refers to: Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs:137 in f6fc3f5. [](commit_id = f6fc3f579ab923125d350149a552ce2274edcb40, deletion_comment = False)

shaopeng-gh commented:
>         private Run ConvertToSarif(List<CweCsvRecord> records, string version)

we have csv code and xml code, currently csv code is not used and I think it is best to remove it, and if we need it we can still get the code form history.
I have removed it.
also I see there are 3 wanrings in the code, I used VS suggested fix and fixed them.

---
In reply to: [868445293](https://github.com/sarif-standard/taxonomies/pull/14#issuecomment-868445293) [](http://example.com/codeflow?ancestors=868445293,868445293)

---
Refers to: Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs:137 in f6fc3f5. [](commit_id = f6fc3f579ab923125d350149a552ce2274edcb40, deletion_comment = False)

shaopeng-gh committed:
remove unused CSV code

eddynaka reviewed (approved):
:shipit:
